### PR TITLE
Fix type conversion of optional types

### PR DESCRIPTION
**Pull request by rnv812**

### CODE REVIEW

**Review by felix-martel — CHANGES_REQUESTED** [review]
Thanks, can you just add a test case for `int | None` and `float | None`? You can duplicate `test_get_type_from_field_with_unconstrained_int` from `tests/test_type_conversion.py`.

File: pydanclick/model/type_conversion.py [hunk 1]
```diff
@@ -168,7 +169,7 @@ def _get_click_type_from_field(field: FieldInfo) -> click.ParamType:
     field_origin = get_origin(field_type)
     # TODO: handle annotated
     # TODO: handle subclasses
-    if field_origin is Union and len(field_args) == 2 and NoneType in field_args and field.default is None:
+    if field_origin is UnionType and len(field_args) == 2 and NoneType in field_args:
```
Comment (felix-martel) [inline pydanclick/model/type_conversion.py:172]: `types.UnionType` was introduced in Python 3.10

Comment (rnv812) [reply pydanclick/model/type_conversion.py]: Add check of python version, so now comparing works well both for 3.9 and later versions

File: tests/test_type_conversion.py [hunk 2]
```diff
@@ -17,6 +17,22 @@ class Foo(BaseModel):
     assert isinstance(click_type, click.types.IntParamType)
 
 
+def test_get_type_from_field_with_optional_int():
+    class Foo(BaseModel):
+        a: int | None = None
```
Comment (felix-martel) [inline tests/test_type_conversion.py:22]: Can't use `|` on Python 3.9 unfortunately...

```suggestion
        a: Optional[int] = None
```

File: tests/test_type_conversion.py [hunk 3]
```diff
@@ -17,6 +17,22 @@ class Foo(BaseModel):
...
+
+    click_type = _get_type_from_field(Foo.model_fields["a"])
+    assert isinstance(click_type, click.types.IntParamType)
+
+
+def test_get_type_from_field_with_optional_float():
+    class Foo(BaseModel):
+        a: float | None = None
```
Comment (felix-martel) [inline tests/test_type_conversion.py:30]:
```suggestion
        a: Optional[float] = None
```

**Review by felix-martel — CHANGES_REQUESTED** [review]
LGTM, just apply `ruff` and fix tests on Python 3.9, then we're good to go!

File: tests/test_type_conversion.py (hunk 3 above)
Comment (rnv812) [reply tests/test_type_conversion.py]: Ahh, you're right, my bad :)

**Comment by rnv812** [comment]
Formatting is fixed

File: tests/test_type_conversion.py [hunk 4]
```diff
@@ -17,6 +17,22 @@ class Foo(BaseModel):
     assert isinstance(click_type, click.types.IntParamType)
 
 
+def test_get_type_from_field_with_optional_int():
+    class Foo(BaseModel):
+        a: Optional[int] = None
```
Comment (felix-martel) [inline tests/test_type_conversion.py:22]: You need `from typing import Optional`

**Review by felix-martel — CHANGES_REQUESTED** [review]

**Comment by rnv812** [comment]
Fix problems with `pre-commit run --all-files`

**Review by felix-martel — APPROVED** [review]
LGTM!
